drivers/gpio/gpio_mcux.c

```diff
@@ -404,47 +445,44 @@ static DEVICE_API(gpio, gpio_mcux_driver_api) = {
 #endif /* CONFIG_GPIO_GET_DIRECTION */
 };
 
-#define GPIO_MCUX_IRQ_INIT(n)						\
-	do {								\
-		IRQ_CONNECT(DT_INST_IRQN(n),				\
-			    DT_INST_IRQ(n, priority),			\
-			    gpio_mcux_port_isr,				\
-			    DEVICE_DT_INST_GET(n), 0);			\
-									\
-		irq_enable(DT_INST_IRQN(n));				\
+#define GPIO_MCUX_IRQ_INIT(n)                                                                      \
+	do {                                                                                       \
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority), gpio_mcux_port_isr,         \
+			    DEVICE_DT_INST_GET(n), 0);                                             \
+                                                                                                   \
+		irq_enable(DT_INST_IRQN(n));                                                       \
 	} while (false)
 
 #define GPIO_PORT_BASE_ADDR(n) DT_REG_ADDR(DT_INST_PHANDLE(n, nxp_kinetis_port))
-
-#define GPIO_DEVICE_INIT_MCUX(n)					\
-	static int gpio_mcux_port## n ## _init(const struct device *dev); \
-									\
-	static const struct gpio_mcux_config gpio_mcux_port## n ## _config = {\
-		.common = {						\
-			.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(n),\
-		},							\
-		.gpio_base = (GPIO_Type *) DT_INST_REG_ADDR(n),		\
-		.port_base = (PORT_Type *) GPIO_PORT_BASE_ADDR(n),	\
-		.flags = UTIL_AND(UTIL_OR(DT_INST_IRQ_HAS_IDX(n, 0),	\
-				GPIO_HAS_SHARED_IRQ), GPIO_INT_ENABLE), \
-	};								\
-									\
-	static struct gpio_mcux_data gpio_mcux_port## n ##_data;	\
-									\
-	DEVICE_DT_INST_DEFINE(n,					\
-			    gpio_mcux_port## n ##_init,			\
-			    NULL,					\
-			    &gpio_mcux_port## n ##_data,		\
-			    &gpio_mcux_port## n##_config,		\
-			    POST_KERNEL,				\
-			    CONFIG_GPIO_INIT_PRIORITY,			\
-			    &gpio_mcux_driver_api);			\
-									\
-	static int gpio_mcux_port## n ##_init(const struct device *dev)	\
-	{								\
+#define GPIO_PORT_NUMBER(n) COND_CODE_1(DT_INST_NODE_HAS_PROP(n, gpio_port_offest),	\
```

Comment:
If possible splitting the formatting changes here (which seem to be from running clang-format) out from the functional changes to the driver would make this easier to review. If not, just be aware that is best practice in the future (not to mix formatting changes and functional ones in the same commit)

Reply:
Split the format changes. Thank you..